hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1240 @@
+        every 1min scheduler will call this method and check Status of
+        room is occupied or available
+        --------------------------------------------------------------
+        @param self: The object pointer
+        @return: update status of hotel room reservation line
+        """
+        reservation_line_obj = self.env["hotel.room.reservation.line"]
+        folio_room_line_obj = self.env["folio.room.line"]
+        now = datetime.now()
+        curr_date = now.strftime(dt)
+        for room in self.search([]):
+            reserv_line_ids = [
+                reservation_line.id
+                for reservation_line in room.room_reservation_line_ids
+            ]
+            reservation_line_ids = reservation_line_obj.search(
+                [
+                    ("id", "in", reserv_line_ids),
+                    ("check_in", "<=", curr_date),
+                    ("check_out", ">=", curr_date),
+                ]
+            )
+            rooms_ids = [room_line.id for room_line in room.room_line_ids]
+            room_line_ids = folio_room_line_obj.search(
+                [
+                    ("id", "in", rooms_ids),
+                    ("check_in", "<=", curr_date),
+                    ("check_out", ">=", curr_date),
+                ]
+            )
+            status = {"isroom": True, "color": 5}
+            if reservation_line_ids:
+                status = {"isroom": False, "color": 2}
+            room.write(status)
+            if room_line_ids:
+                status = {"isroom": False, "color": 2}
+            room.write(status)
+            if reservation_line_ids and room_line_ids:
+                raise ValidationError(
+                    _("Please Check Rooms Status for %s.") % room.name
+                )
+        return True
+
+
+class RoomReservationSummary(models.Model):
+
+    _name = "room.reservation.summary"
+    _description = "Room reservation summary"
+
+    @api.model
+    def _default_date_from(self):
+        return datetime.today()
+
+    @api.model
+    def _default_date_to(self):
+        return datetime.today() + relativedelta(days=30)
+
+    name = fields.Char(
+        "Reservation Summary", default="Reservations Summary", invisible=True
+    )
+    date_from = fields.Datetime("Date From", default=_default_date_from)
+    date_to = fields.Datetime("Date To", default=_default_date_to)
+    summary_header = fields.Text("Summary Header")
+    room_summary = fields.Text("Room Summary")
+
+    def room_reservation(self):
+        """
+        @param self: object pointer
+        """
+        mod_obj = self.env["ir.model.data"]
+        if self._context is None:
+            self._context = {}
+        model_data_ids = mod_obj.search(
+            [
+                ("model", "=", "ir.ui.view"),
+                ("name", "=", "view_hotel_reservation_form"),
+            ]
+        )
+        resource_id = model_data_ids.read(fields=["res_id"])[0]["res_id"]
```

Comment:
Please remove this one and simply use self.env.ref()

Reply:
Done.